app/controllers/advertisements_controller.rb

```diff
@@ -21,10 +21,21 @@ def set_virtual_impression_id
     @virtual_impression_id = SecureRandom.uuid
   end
 
+  def ip_info
+    @ip_info ||= MMDB.lookup(request.remote_ip)
+  end
+
   def set_campaign
-    # TODO: smarter campaign selection
-    @campaign = Campaign.active.available_on(Date.current).for_property_id(params[:property_id]).limit(10).to_a.sample
-    @campaign ||= Campaign.active.available_on(Date.current).fallback_for_property_id(params[:property_id]).limit(10).to_a.sample
+    country_code = ip_info.country.iso_code
```

Comment:
will `MMDB.lookup` always return results even if it's the IP is bad or can't be found?

Reply:
Yes but the result may be an "empty" result but it will behave just like any other result.